lib/shrine/plugins/validate_correct_form.rb

```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+
+require 'common/file_helpers'
+
+class Shrine
+  module Plugins
+    module ValidateCorrectForm
+      module AttacherMethods
+        WRONG_FORM = 'wrong_form'
+
+        def validate_correct_form(form_id: nil)
+          return unless get.mime_type == Mime[:pdf].to_s && form_id
+
+          image_path = Rails.root.join("#{Common::FileHelpers.random_file_path}.jpg").to_s
+          file = get.download
+          pdf = MiniMagick::Image.open(file.path)
+          MiniMagick::Tool::Convert.new do |convert|
+            convert.background 'white'
+            convert.flatten
+            convert.density 150
+            convert.quality 100
+            convert << pdf.pages.first.path
+            convert << image_path
+          end
+          file_as_string = RTesseract.new(image_path).to_s
+
+          record.warnings << WRONG_FORM unless file_as_string.include? form_id
```

Comment:
Out of curiosity, what does this do and how do we use these warnings?

Reply:
Great question! This is a very naive, first-step towards using OCR to determine if the pdf is the right form. So at the moment it will convert the form to a string and check that string for appearances of the form id. [We put the warning on the FE.](https://github.com/department-of-veterans-affairs/vets-website/commit/46dec6716562d03be36a1039456d4f929ba5cb14)